### PR TITLE
Preserve mark_dynamic when cloning inputs

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #99634
* __->__ #99617

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

### CODE REVIEW

**Review by ngimel — APPROVED** [review]

**Review by voznesenskym — APPROVED** [review]

File: benchmarks/dynamo/ci_expected_accuracy/inductor_huggingface_dynamic_inference.csv [hunk 1]
```diff
@@ -1,43 +1,43 @@
 name,accuracy,graph_breaks
-AlbertForMaskedLM,pass,0
-AlbertForQuestionAnswering,pass,0
-BartForCausalLM,pass,0
-BertForMaskedLM,pass,0
-BertForQuestionAnswering,pass,0
+AlbertForMaskedLM,fail_to_run,0
```
Comment (ngimel) [inline benchmarks/dynamo/ci_expected_accuracy/inductor_huggingface_dynamic_inference.csv:2]: ouch, but they used to run before we staticified everything?

Comment (ezyang) [reply benchmarks/dynamo/ci_expected_accuracy/inductor_huggingface_dynamic_inference.csv:2]: So, this is more stringent than what we used to have before. Prior to static everything, we yolo dynamic everything, but if we FAILED to make it dynamic, we would just silently go back to static perf. So this is more strict than anything we've ever had in the benchmark script.

BTW, I think most of the HF failures are due to just some overspecialization on slice. Just need to fix it.

**Review by anijain2305 — APPROVED** [review]

**Comment by ezyang** [comment]
merging this pr is suffering

**Pull request closed without merging**